### PR TITLE
Fix bugs around creating ECDAA key

### PR DESCRIPTION
**Pull request by akakou**
There are the following bugs around creating ECDAA key:

-   CurveBNP256 number is wrong
    -  TPM expects `0x10` to `TPM2_ECC_BN_P256` but this library does not set so.
-  ECDAA `count` (in `TPMS_SCHEME_ECDAA`) size is wrong
    -  TPM expects 16bit but, actually 32bit in go-tpm.

So this PR fixes these bugs.

### Reference

- https://trustedcomputinggroup.org/wp-content/uploads/TSS_Overview_Common_Structures_Version-0.9_Revision-03_Review_030918.pdf

### CODE REVIEW

**Review by chrisfenner — CHANGES_REQUESTED** [review]
This change looks good, only a small fix for CurveBNP638.

BTW, we are doing a major refactor to the API in https://github.com/google/go-tpm/tree/tpmdirect - have you thought about how ECDAA could be exposed via the legacy command model, or do you want to build this in TPMDirect?

File: tpm2/constants.go [hunk 1]
```diff
@@ -426,7 +426,7 @@ const (
 	CurveNISTP384
 	CurveNISTP521
 
-	CurveBNP256 = EllipticCurve(iota + 10)
+	CurveBNP256 = EllipticCurve(0x10)
 	CurveBNP638
```
[the remark this answers is not in the document]
Comment (akakou) [reply tpm2/constants.go:430]: Did you mean the `CurveBNP638`?

[the remark this answers is not in the document]
Comment (akakou) [reply tpm2/constants.go:430]: @chrisfenner

[the remark this answers is not in the document]
Comment (akakou) [reply tpm2/constants.go]: I replaced them on a new commit!

**Comment by akakou** [comment]
Umm...I think we have to implement some TPM commands(e.g. TPM2_Commit) first to run ECDAA. (I made this Issue #287)
Now I'm trying to implement `TPM2_Commit` to the `tpmdirect` branch.

But I don't think that I develop ECDAA API (e.g. ECDAA_JOIN) directly, because there is some process that should not be in the scope of the TPM.
(FIDO ECDAA defines the other module running on a client named ASM. There are some processes for ECDAA running on ASM, not on TPM)

https://fidoalliance.org/specs/fido-v2.0-id-20180227/fido-ecdaa-algorithm-v2.0-id-20180227.html#ecdaa-sign-split-between-tpm-and-asm

**Comment by chrisfenner** [comment]
Sounds good. Do you want to send a PR to the `tpmdirect` branch instead of `master`?

**Comment by akakou** [comment]
You are right! I will change the destination branch to `tpmdirect`.

**Comment by akakou** [comment]
But I am concerned that the master branch remains wrong about BN256 and ECDAA.
Because some people using BN256 or ECDAA with master branch may be confused.

I would like to confirm just in case; the tpmdirect branch will be merged with the master branch finally, right?

**Comment by akakou** [comment]
This discussion was solved because we decided that contribute to `tpmdirect`.
So I close this PR.

**Pull request closed without merging**